src/modules/wreck/job.c

```diff
@@ -608,20 +608,23 @@ static void runevent_cb (flux_t *h, flux_msg_handler_t *w,
     Jput (in);
 }
 
-struct flux_msg_handler_spec mtab[] = {
-    { FLUX_MSGTYPE_REQUEST, "job.create", job_request_cb, 0, NULL },
-    { FLUX_MSGTYPE_REQUEST, "job.submit", job_request_cb, 0, NULL },
-    { FLUX_MSGTYPE_REQUEST, "job.shutdown", job_request_cb, 0, NULL },
-    { FLUX_MSGTYPE_REQUEST, "job.kvspath",  job_kvspath_cb, 0, NULL },
-    { FLUX_MSGTYPE_EVENT,   "wrexec.run.*", runevent_cb, 0, NULL },
+static const struct flux_msg_handler_spec mtab[] = {
+    { FLUX_MSGTYPE_REQUEST, "job.create", job_request_cb, 0 },
+    { FLUX_MSGTYPE_REQUEST, "job.submit", job_request_cb, 0 },
+    { FLUX_MSGTYPE_REQUEST, "job.shutdown", job_request_cb, 0 },
+    { FLUX_MSGTYPE_REQUEST, "job.kvspath",  job_kvspath_cb, 0 },
+    { FLUX_MSGTYPE_EVENT,   "wrexec.run.*", runevent_cb, 0 },
     FLUX_MSGHANDLER_TABLE_END
 };
 
 int mod_main (flux_t *h, int argc, char **argv)
 {
-    if (flux_msg_handler_addvec (h, mtab, NULL) < 0) {
+    flux_msg_handler_t **handlers = NULL;
+    int rc = -1;
+
+    if (flux_msg_handler_addvec (h, mtab, NULL, &handlers) < 0) {
         flux_log_error (h, "flux_msg_handler_addvec");
-        return (-1);
+        goto done;
```

Comment:
This jumps to the wrong place.  If flux_msg_handler_addvec() failed, we shouldn't be calling flux_msg_handler_delvec().

Reply:
delvec is a no-op so it's not wrong per se.  Sometimes it's nice to have a single exit path rather than returns sprinkled throughout a function.    I could go either way here but I'll do as you suggest.